Source/WPE/Source/Input/LibinputServer.cpp

```diff
@@ -204,6 +230,99 @@ void LibinputServer::dispatchKeyboardEvent(const Input::KeyboardEvent::Raw& even
     m_client->handleKeyboardEvent({ event.time, std::get<0>(result), std::get<1>(result), !!event.state, std::get<2>(result) });
 }
 
+
+void LibinputServer::handleTouchEvent(struct libinput_event *event, Input::TouchEvent::Type type, bool has_position)
+{
+    auto* touchEvent = libinput_event_get_touch_event(event);
+    uint32_t time = libinput_event_touch_get_time(touchEvent);
+    int id = libinput_event_touch_get_slot(touchEvent);
+    auto& targetPoint = m_touchEvents[id];
+    int32_t x, y;
+    
+    if (has_position) {
+      x = libinput_event_touch_get_x_transformed(touchEvent, m_touchSize.first);
+      y = libinput_event_touch_get_y_transformed(touchEvent, m_touchSize.second);
+    } else {
+      x = targetPoint.x;
+      y = targetPoint.y;
+    }
+    targetPoint = Input::TouchEvent::Raw{ type, time, id, x, y };
+    
+    m_client->handleTouchEvent({
+      m_touchEvents,
+      type,
+      id,
+      time
+    });
+    
+    if (type == Input::TouchEvent::Down) {
```

Comment:
this block is related to click simulation. it will check if user dont move his finger while touching screen.